rex/mutli_year_resource.py

```diff
@@ -0,0 +1,834 @@
+# -*- coding: utf-8 -*-
+"""
+Classes to handle resource accross multiple files
+"""
+import numpy as np
+import os
+from warnings import warn
+
+from rex.renewable_resource import NSRDB, SolarResource, WindResource
+from rex.resource import MultiH5, Resource
+from rex.utilities.exceptions import ResourceKeyError, ResourceWarning
+from rex.utilities.parse_keys import parse_keys, parse_slice
+from rex.utilities.utilities import parse_year
+
+
+class MultiYearH5:
+    """
+    Class to handle multiple years of h5 Resources
+    """
+
+    def __init__(self, h5_dir, prefix='', suffix='.h5', res_cls=Resource,
+                 **res_cls_kwargs):
+        """
+        Parameters
+        ----------
+        h5_dir : str
+            Path to directory containing 5min .h5 files
+        prefix : str
+            Prefix for resource .h5 files
+        suffix : str
+            Suffix for resource .h5 files
+        res_cls : obj
+            Resource class to use to open and access resource data
+        """
+        self.h5_dir = h5_dir
+        self._year_map = self._map_file_years(h5_dir, prefix=prefix,
+                                              suffix=suffix)
+        self._h5_map = self._map_file_instances(set(self._year_map.values()),
+                                                res_cls=res_cls,
+                                                **res_cls_kwargs)
+        self._datasets = None
+        self._shape = None
+        self._time_index = None
+        self._i = 0
+
+    def __repr__(self):
+        msg = ("{} for {}:\n Contains data for {} year"
+               .format(self.__class__.__name__, self.h5_dir, len(self)))
+        return msg
+
+    def __len__(self):
+        return len(self.years)
+
+    def __getitem__(self, year):
+        if isinstance(year, str):
+            year = int(year)
+
+        if year in self.years:
+            path = self._year_map[year]
+            h5 = self._h5_map[path]
+        else:
+            raise ValueError('{} is invalid, must be one of: {}'
+                             .format(year, self.years))
+
+        return h5
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self._i >= len(self.years):
+            self._i = 0
+            raise StopIteration
+
+        year = self.years[self._i]
+        self._i += 1
+
+        return year
+
+    def __contains__(self, year):
+
+        return year in self.years
+
+    @property
+    def attrs(self):
```

Comment:
Can we add this to the normal Resource class? Can't tell you how many times i've wanted it instead of calling .get_attrs()

Reply:
Sure